compiler/rustc_codegen_llvm/src/back/lto.rs

```diff
@@ -604,7 +604,12 @@ pub(crate) fn run_pass_manager(
     debug!("running the pass manager");
     let opt_stage = if thin { llvm::OptStage::ThinLTO } else { llvm::OptStage::FatLTO };
     let opt_level = config.opt_level.unwrap_or(config::OptLevel::No);
-    unsafe { write::llvm_optimize(cgcx, dcx, module, config, opt_level, opt_stage) }?;
+    // We will run this again with different values in the context of automatic differentiation.
+    let first_run = true;
```

Comment:
Is this happening regardless of whether Enzyme is enabled?

Reply:
No, only if Enzyme support is enabled when building rustc, and if the user additionally applied at least one autodiff macro.
If that is the case we check that fat-lto is enabled (in the other PR), and then we will run opt passes twice.
The fat-lto requirement is something that will be lifted in the future. 
Optimizing the whole module twice is also a bit more then what we really want, so I'll make that more granular in the future, to only optimize code that is getting differentiated twice. I'll add this to the docs.